mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1349,17 +1381,19 @@ impl TakerSwap {
         let tx_hex = self.r().taker_payment.as_ref().unwrap().tx_hex.0.clone();
         let mut watcher_broadcast_abort_handle = None;
         if self.ctx.use_watchers() {
-            let preimage_hex = self.r().taker_spends_maker_payment_preimage.clone();
-            if let Some(preimage_hex) = preimage_hex {
-                let watcher_data = self.create_watcher_data(tx_hex.clone(), preimage_hex);
-                let swpmsg_watcher = SwapWatcherMsg::TakerSwapWatcherMsg(Box::new(watcher_data));
-                watcher_broadcast_abort_handle = Some(broadcast_swap_message_every(
-                    self.ctx.clone(),
-                    watcher_topic(&self.r().data.taker_coin),
-                    swpmsg_watcher,
-                    600.,
-                    self.p2p_privkey,
-                ));
+            if let Some(taker_spends_maker_payment) = self.r().taker_spends_maker_payment_preimage.clone() {
+                if let Some(taker_refunds_payment) = self.r().taker_refunds_payment.clone() {
+                    let watcher_data =
+                        self.create_watcher_data(tx_hex.clone(), taker_spends_maker_payment, taker_refunds_payment);
+                    let swpmsg_watcher = SwapWatcherMsg::TakerSwapWatcherMsg(watcher_data);
+                    watcher_broadcast_abort_handle = Some(broadcast_swap_message_every(
+                        self.ctx.clone(),
+                        watcher_topic(&self.r().data.taker_coin),
+                        swpmsg_watcher,
+                        600.,
```

Comment:
Could you please create a in-function constant for this? Like `MAKER_PAYMENT_WAIT_TIMEOUT`.

It makes more sense on reading the code-block. Also used twice in same function.